Enable multi-selection on Index component

huiwang-hpe opened the pull request:
Our App use Grommet-Index to show the data.
And we have a new requirement that user can select multiple items if he/she switches to the selection mode.
However, Grommet-Index only support single selection.

Luckily, the Tiles, List and Table do support multi-selection.
So I updated the interface to accept not only a single string URI, but also a array of URIs.

This won't change the existing interface but only enhanced its capabilities.

BTW, the babel-polyfill is referenced as I used Array.includes() and Array.isArray().

huiwang-hpe committed:
Enable multi-selection on Index component

huiwang-hpe committed:
Merge remote-tracking branch 'upstream/master'

alansouzati commented on src/js/components/List.js:
```diff
@@ -90,11 +90,16 @@ export default class IndexList extends Component {
 
     let listItems;
-    let selectionIndex;
+    let multiSelected = Array.isArray(selection);
```
i believe this can be const instead of let

alansouzati commented on src/js/components/List.js:
```diff
@@ -90,11 +90,16 @@ export default class IndexList extends Component {
     let listItems;
-    let selectionIndex;
+    let multiSelected = Array.isArray(selection);
+    let selectionIndex = multiSelected ? [] : null;
```
can we use undefined instead of null ?

alansouzati reviewed (commented):
I like the changes, thanks for working on those and sorry for taking such a long time to get back to them.

Can you apply the minor feedbacks I gave? it would be great if you can propagate the changes to Table and Tiles as well.

alansouzati commented:
Also can you squash your commit? We are trying to avoid things like `Merge remote-tracking branch 'upstream/master'` in the change log.

huiwang-hpe committed:
Update for multi-selection feature

huiwang-hpe commented:
@alansouzati I updated codes according to your comments. However, I forked grommet-index at 0.4.5 and merged 1.0.0 into it. Now I don't know how to squash it because there are other commits between my commits. I added you as the contributor and please check if you can help. Thanks.

alansouzati commented:
Don't worry about the squash, thanks for your contribution.

For next time, I would recommend watching this video where I explain the contribution process, including squashing commits: https://vimeo.com/129681048
